presto-geospatial-toolkit/src/main/java/com/facebook/presto/geospatial/JtsGeometrySerde.java

```diff
@@ -358,58 +245,110 @@ private static void writePolyline(Geometry geometry, SliceOutput output, boolean
         return coordinates;
     }
 
-    private static Geometry readPolygon(SliceInput input)
+    /**
+     * Serialize JTS {@link Geometry} shape into an ESRI shape
+     */
+    public static Slice serialize(Geometry geometry)
     {
-        requireNonNull(input, "input is null");
-        skipEnvelope(input);
-        int partCount = input.readInt();
-        if (partCount == 0) {
-            return GEOMETRY_FACTORY.createPolygon();
+        requireNonNull(geometry, "input is null");
+        DynamicSliceOutput output = new DynamicSliceOutput(100);
+        writeGeometry(output, geometry);
+        return output.slice();
+    }
+
+    private static void writeGeometry(DynamicSliceOutput output, Geometry geometry)
+    {
+        switch (geometry.getGeometryType()) {
+            case "Point":
+                writePoint((Point) geometry, output);
+                break;
+            case "MultiPoint":
+                writeMultiPoint((MultiPoint) geometry, output);
+                break;
+            case "LineString":
+            case "MultiLineString":
+                writePolyline(geometry, output);
+                break;
+            case "Polygon":
+            case "MultiPolygon":
+                writePolygon(geometry, output);
+                break;
+            case "GeometryCollection":
+                writeGeometryCollection(output, geometry);
```

Comment:
Can you swap the arguments to `writeGeometryCollection` so it becomes `writeGeometryCollection(geometry, output)` to make it consistent with the other write methods.